src/components/timeline/LayerDiscrete.svelte

```diff
@@ -404,76 +462,108 @@
   }
 
   function drawGroupedMode() {
+    // must clear the canvas before a redraw! otherwise the old ungrouped version can linger around and we draw over that!
+    canvas.getContext('2d')?.clearRect(0, 0, canvas.width, canvas.height);
     if (xScaleView !== null) {
-      const collapsedMode = drawHeight <= ViewConstants.MIN_ROW_HEIGHT;
-      let y = collapsedMode ? 0 : ViewConstants.MIN_ROW_HEIGHT - 1; // pad starting y with the min row height to align with activity tree
-      const expectedRowHeight = rowHeight + activityRowPadding;
-      activityTree.forEach(node => {
-        const newY = drawGroup(node, y, expectedRowHeight, !collapsedMode);
-        if (!collapsedMode) {
+      // expanded cannot possibly be false
+      let y = !expanded ? 0 : ViewConstants.MIN_ROW_HEIGHT - 1; // pad starting y with the min row height to align with activity tree
+      const expectedRowHeight = rowHeight + discreteRowPadding;
+
+      discreteTree.forEach(node => {
+        // TODO: add a gap between activities and external events if both are present together.
+        const newY = drawGroup(node, y, expectedRowHeight, expanded);
+        if (expanded) {
           y = newY;
         }
       });
+
       const newRowHeight = y + 36; // add padding to the bottom to account for buttons in the activity tree
-      if (!collapsedMode && newRowHeight > 0) {
+      if (expanded && newRowHeight > 0) {
         /* TODO a change from manual to auto height does not take effect until you trigger a redraw on this row, could pass in whether or not to update row height but that might be odd? */
         dispatch('updateRowHeight', { newHeight: newRowHeight });
       }
     }
   }
 
+  // TODO: as height reduces, merge items into the same rows, and make labels disappear. This might require significantly reworking this method.
```

Comment:
Not sure I understand the idea behind this comment, can you elaborate?

Reply:
As the row height is dragged and reduced, the only thing that changes is the gap between items, which produces an intermediary result like what follows:
<img width="949" alt="image" src="https://github.com/user-attachments/assets/bda82715-b360-464a-b369-407b99245183">

All labels are rendered, and rows that are initially somewhat distinguished start to merge in a continuous fashion, like the image shows. The idea behind this comment is that _at some point_ this should be refactored (this behavior is also present in the UI outside of this commit, in an activity-only row).

Reply:
So a little history here – we did have some additional item packing logic on the timeline for a while similar to what you're proposing but it proved to be not the solution users were looking for. In the end we purposefully went for this less automated approach that other tools have used and users have requested, meaning users get/have to be in control of the height/collision which obviously means you can get into situations like this. There are improvements we can make to this system like responding to auto height on the row (causes row height to resize a lot which can be jarring though), a button to auto or one time distribute height row between activities, a mode to vertically pack activities upwards and only collapse them further when space is lacking, etc etc. But I don't think we're looking to go back to auto moving elements if they vertically collide.

Reply:
So this behavior, as is, is ideal and expected behavior and the comment can be removed?

Reply:
Per a later slack call, it was stated clearly that this is ideal behavior. In cases like compacting a row for a screenshot, this can actually be very desirable. As such, no action is needed!